### ISSUE TITLE
Support Strapi 5

### ISSUE THREAD

**Issue by xiezhuangping** [ChristopheCVB/strapi-plugin-soft-delete#15]
**Is your feature request related to a problem? Please describe.**

Can not install in strapi 5.
![image](https://github.com/user-attachments/assets/bdbf68fd-2941-44dd-ac62-fddb02ef3888)

`
strapi>npm install strapi-plugin-soft-delete
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: strapi@0.1.0
npm ERR! Found: @strapi/strapi@5.0.1
npm ERR! node_modules/@strapi/strapi
npm ERR!   @strapi/strapi@"5.0.1" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @strapi/strapi@"^4.11.0" from strapi-plugin-soft-delete@0.9.6
npm ERR! node_modules/strapi-plugin-soft-delete
npm ERR!   strapi-plugin-soft-delete@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See C:\Users\fab\AppData\Local\npm-cache\eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\x\AppData\Local\npm-cache\_logs\2024-10-10T08_38_12_585Z-debug-0.log
`

**Comment by shubibees** [comment]
hi, same issue

**Comment by carlo-pi** [comment]
same here. @ChristopheCVB is an update planned?

**Comment by anthonytamez** [comment]
+1

@ChristopheCVB is there a plan for v5 support?

**Comment by ChristopheCVB** [comment]
When/If I find the time, yeah :D
PRs are welcome of course

**Comment by jmichalec-vl** [comment]
@ChristopheCVB [here](https://github.com/ChristopheCVB/strapi-plugin-soft-delete/pull/16) is PR with upgrade to strapi v5 proposal

**Comment by carlo-pi** [comment]
@ChristopheCVB can you please check @jmichalec-vl PR?

**Comment by anthonytamez** [comment]
@ChristopheCVB  is there a planned timeline for @jmichalec-vl PR review/acceptance?

**Comment by jmichalec-vl** [comment]
I can't wait any longer for maintainer to review and merge my PR,  I [created plugin version for v5](https://github.com/jmichalec-vl/strapi-soft-delete-plugin). Also I added features which was missing in v4 plugin version. Plugin is waiting for review from strapi team to be added in strapi market place, feel free to add issues if u find them. I checked this plugin on one project where it work, but as always I could miss some cases.

fyi @anthonytamez, @carlo-pi, @shubibees, @xiezhuangping

**Comment by ChristopheCVB** [comment]
> I can't wait any longer for maintainer to review and merge my PR, I [created plugin version for v5](https://github.com/jmichalec-vl/strapi-soft-delete-plugin?rgh-link-date=2026-05-15T10%3A53%3A12.000Z). Also I added features which was missing in v4 plugin version. Plugin is waiting for review from strapi team to be added in strapi market place, feel free to add issues if u find them. I checked this plugin on one project where it work, but as always I could miss some cases.
> 
> fyi [@anthonytamez](https://github.com/anthonytamez), [@carlo-pi](https://github.com/carlo-pi), [@shubibees](https://github.com/shubibees), [@xiezhuangping](https://github.com/xiezhuangping)

@jmichalec-vl thanks for taking over, I really didn't have the time to maintain this anymore

**Comment by ChristopheCVB** [comment]
Also, seems to be a problem with building your plugin. dist directory ends up being 2.11MB, weird no?
https://www.npmjs.com/package/strapi-soft-delete-plugin?activeTab=code

<img width="1195" height="524" alt="Image" src="https://github.com/user-attachments/assets/b1b38985-cf8f-4ddc-b18d-f6704e7d2f36" />

**Comment by jmichalec-vl** [comment]
Good catch! The issue was @strapi/admin missing from the dependencies list — the bundler was inlining Radix UI primitives from @strapi/design-system instead of externalizing them. Fixed — dist went from 2.1MB to 180KB. Will publish an updated version shortly.
